src/background/vmModules/ApprovalController.ts

```diff
@@ -175,7 +175,11 @@ export class ApprovalController implements IApprovalController {
       case RpcMethod.BITCOIN_SEND_TRANSACTION:
       case RpcMethod.BITCOIN_SIGN_TRANSACTION:
       case RpcMethod.ETH_SEND_TRANSACTION:
-        return await this.#walletService.sign(signingData.data, network);
+        return await this.#walletService.sign(
+          signingData.data,
+          network,
+          action.tabId
```

Comment:
Pass `tabId`, so the signature request can later be paired with the window that initiated the request